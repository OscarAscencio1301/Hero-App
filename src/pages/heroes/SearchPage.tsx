import { FormEvent, useMemo } from "react"
import useForm from "../../hooks/useForm"
import { useSearchParams } from "react-router-dom"
import { getHeoresbyQuery } from "../../helpers/getHeroebyQuery"
import HeroeCard from "../../components/HeroeCard"


const SearchPage = () => {
  const [searchParams, setsearchParams] = useSearchParams()

  const { search, changeValues } = useForm({
    search: ''
  })


  const sendFormSearch = (e: FormEvent) => {
    e.preventDefault()

    if (search.trim().length < 1) return

    setsearchParams({ q: search.toString() })


  }


  const query = searchParams.get('q')
  const heroes = useMemo(() => getHeoresbyQuery(query?.toString()), [query])




  return (
    <>
      <h1>Search</h1>
      <hr />
      <div className="row">
        <div className="col-5">
          <h4>Searching</h4>
          <hr />
          <form onSubmit={sendFormSearch}>
            <input type="text" placeholder="Search Hero" className="form-control" name="search" autoComplete="off" value={search} onChange={changeValues} />
            <button type="submit" className="btn mt-2 btn-primary">Search</button>
          </form>
        </div>
        <div className="col-7">
          <h4>Results</h4>
          <hr />

          {
            heroes?.map(heroe => <HeroeCard {...heroe} key={heroe.id} />)
          }
        </div>
      </div>
    </>
  )
}


export default SearchPage
