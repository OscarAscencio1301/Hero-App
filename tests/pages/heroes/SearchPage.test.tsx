import { render, screen } from "@testing-library/react";
import SearchPage from "../../../src/pages/heroes/SearchPage";
import { MemoryRouter } from "react-router-dom";

describe('Testing a Search Page', () => {
    test('Snapshot ', () => {
        const { container } = render(
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
        )
        expect(container).toMatchSnapshot()
    });

    test('Debe mostrar el resultado con el valor en el input ', () => {
        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage />
            </MemoryRouter>
        )
        const img = screen.getByRole('img') as HTMLImageElement
        expect(img.src).toContain("/assets/heroes/dc-batman.jpg")

        const titleResult = screen.getByLabelText('text-results')
        expect(titleResult.style.display).toBe('none')


    
    });
})
