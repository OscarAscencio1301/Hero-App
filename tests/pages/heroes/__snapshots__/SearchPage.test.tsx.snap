// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing a Search Page Snapshot  1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h3>
        Searching
      </h3>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="search"
          placeholder="Search Hero"
          type="text"
          value=""
        />
        <button
          class="btn mt-2 btn-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4
        aria-label="text-results"
        style="display: undefined;"
      >
        Results
      </h4>
      <hr />
    </div>
  </div>
</div>
`;
